mlir/lib/Dialect/MIOpen/Transforms/ThreadwiseToStdlib.cpp

```diff
@@ -246,18 +246,51 @@ struct BufferLoadRewritePattern : public OpRewritePattern<BufferLoadOp> {
         oobDims.push_back(pair.index());
     bool toEmitOobChecks = !oobDims.empty();
 
-    auto emitLoadInstruction = [&b, loc, loadedType,
+    auto emitLoadInstruction = [&b, loc, loadedType, sourceType,
                                 source](ValueRange loadCoords) -> Value {
       Value loadedValue;
-      if (auto loadedVectorType = loadedType.dyn_cast<VectorType>()) {
-        // Issue vector load.
-        // use buffer load since the source memref is on address space 0
-        SmallVector<Value, 4> srcLowerIndicesI32;
-        for (auto v : loadCoords)
-          srcLowerIndicesI32.push_back(
-              b.create<IndexCastOp>(loc, b.getIntegerType(32), v));
-        loadedValue = b.create<gpu::MubufLoadOp>(loc, loadedType, source,
-                                                 srcLowerIndicesI32);
+
+      Type elementType = loadedType;
+      int64_t vectorLength = 1;
```

Comment:
These two lines seems to be used only in `if (loadedType.isa<VectorType>)` path.

Reply:
Make sense will address

Reply:
Done in [d66a280](https://github.com/ROCmSoftwarePlatform/llvm-project-mlir/pull/539/commits/d66a280e92596aa61d7d49e43c1a507880e8a74e)